### PR TITLE
call callback with error instead of throwing

### PR DESCRIPTION
**Pull request by tmcw**
@miccolis do you know how this branch is triggered so we could write test coverage for it?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,27 +2,30 @@ var fontnik = require('fontnik');
...
     var stack = [];
     fontnik.load(opts.font, function(err, faces) {
-        if (err) return callback(err);
+        if (err) {
+            return callback(err);
+        }
 
-        if (faces.length > 1) throw new Error('woah multifont!');
+        if (faces.length > 1) {
+            return new Error('Multiple fonts are not yet supported.');
```
Comment (miccolis) [inline index.js]: Should be return `callback(new Error...))` no?

**Comment by lbud** [comment]
bitmoji

oooooops

> do you know how this branch is triggered so we could write test coverage for it?

The reason there wasn't a test for it was because we didn't know how it was triggered — the [FreeType spec says](https://www.freetype.org/freetype2/docs/reference/ft2-base_interface.html#FT_FaceRec) certain fonts _can_ have multiple faces in one file, but this is rare. @miccolis did the log reveal which font crashed it?

**Comment by miccolis** [comment]
@lbud nope, in typical crashy crash fashion, it just crashed and all I got was this lousy stack trace...

**Comment by tmcw** [comment]
Okay, next commit includes Noto, hits decent coverage numbers, switches to TAP. Unfortunately node-fontnik also has its _own_ throw, so I'm wrapping the call in a try/catch now. Upstream for that is https://github.com/mapbox/node-fontnik/issues/112

File: test/test.js [hunk 2]
```diff
@@ -29,7 +31,31 @@ tape('handle undefined style_name', function(t) {
     });
 });
 
-tape('font machine 256-511', function(t) {
+test('handle font not found', function(t) {
+    fontmachine.makeGlyphs({font: 'NOT-FOUND', filetype: '.ttc'}, function(err, font) {
```
Comment (mikemorris) [inline test/test.js]: So... this is just passing a string directly through to `fontnik.load`, when the API requires passing a `Buffer`. Adding some argument validation (`opts.font instanceof Buffer && opts.callback instanceof Function`) at the beginning of [`makeGlyphs`](https://github.com/mapbox/fontmachine/blob/9aabe5d98f3820296b2712c92777b59add3c2477/index.js#L19) seems like a better approach than `try/catch`  around `fontnik.load` or trying to change this upstream to pass an error through to the callback? If an attempt to read a font from disk fails in a downstream module, shouldn't that error be handled before trying to pass along something invalid to `fontmachine`?

**Comment by mikemorris** [comment]
Clarified the wording of the multi-face font error callback and added some checks that will throw if the API is misused, fix here and switch to `tap` look 💯 to me.

**Pull request merged**